doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,17 @@
+## Make errors
+
+### I'm getting an error when I make in `semgrep-core` after I pull
+
+There are probably changes to submodules that you don't have. Run `git submodule update --recursive`. 
+
+Additionally, `semgrep-core` depends on `spacegrep`, which is a separate binary. If you've already updated submodules, try `make && make install` in `./spacegrep`
```

Comment:
I don't think `semgrep-core` calls the `spacegrep` executable. It's `semgrep` that still does, for now.